File: netcore/PerfTest/Services/MockDataService.cs
using PerfTest.Data.Repositories;
using PerfTest.Domain;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace PerfTest.Services
{
    public class MockDataService
    {
        private readonly MockDataRepository _mockDataRepository;
        private readonly MockDataRepositoryDapper _mockDataRepositoryDapper;

        public MockDataService(MockDataRepository mockDataRepository, MockDataRepositoryDapper mockDataRepositoryDapper)
        {
            _mockDataRepository = mockDataRepository;
            _mockDataRepositoryDapper = mockDataRepositoryDapper;
        }

        public Task<List<MockData>> FindAllByIdSearchAsync(int idSearch)
        {
            return _mockDataRepository.FindAllByIdSearchAsync(idSearch);
        }

        public Task<IEnumerable<MockData>> FindAllByIdSearchAsyncDapper(int idSearch)
        {
            return _mockDataRepositoryDapper.FindAllByIdSearchAsync(idSearch);
        }
    }
}

File: java/src/main/java/com/firstdata/perftest/services/MockDataService.java
package com.firstdata.perftest.services;

import com.firstdata.perftest.domain.MockData;
import com.firstdata.perftest.repositories.MockDataRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MockDataService {
    @Autowired
    private MockDataRepository mockDataRepository;

    public List<MockData> findAllByIdSearch(int idSearch, int type) {
        switch (type) {
            case 1:
            default:
                return mockDataRepository.findAllByIdSearch(idSearch);
            case 2:
                return mockDataRepository.findAllByIdSearchNative(idSearch);
            case 3:
                return mockDataRepository.findAllByIdSearchJdbc(idSearch);
        }
    }
}
